### PR TITLE
[1LP][RFR] Widgetastic ProvisioningForm 

### PR DESCRIPTION
**Pull request by mshriver**
Split this from PR #4013 to get it merged sooner since its imported by several modules that we'll be converting to widgetastic.

I've tested locally with instance provisioning and the form was properly instantiated and filled. I didn't hit every provisioning field in testing.

Leaving the old form in place for now, it can be removed later once we've removed the dependencies on it.

# PRT
Upstream PRT failure  can be ignored. 5.6 and 5.7 passed, basically irrelevant.

### CODE REVIEW

File: cfme/provisioning.py [hunk 1]
```diff
@@ -35,6 +40,137 @@ def select_security_group(sg):
...
+
+    @View.nested
+    class catalog(Tab):  # noqa
+        TAB_NAME = 'Catalog'
+        num_instances = BootstrapSelect('service__number_of_vms')
+        instance_name = Input(name='service__vm_name')
+        instance_description = Input(name='service__vm_description')
+        vm_filter = BootstrapSelect('service__vm_filter')
+        num_vms = BootstrapSelect('service__number_of_vms')
+        vm_name = Input(name='service__vm_name')
```
Comment (otsuman) [inline cfme/provisioning.py]: Maybe instead of "vm_name" and "instance_name" better to add one "name" input? They have same locator.
Same with description

Comment (mshriver) [inline cfme/provisioning.py]: Agreed, I didn't audit the form too much from its existing fields. I'll condense these though.

Comment (mshriver) [inline cfme/provisioning.py]: pushed a 2nd commit, there were a few more duplicates as well in the other tabs.

File: cfme/provisioning.py [hunk 2]
```diff
@@ -35,6 +40,126 @@ def select_security_group(sg):
...
+    @View.nested
+    class properties(Tab):  # noqa
+        TAB_NAME = 'Properties'
+        instance_type = BootstrapSelect('hardware__instance_type')
+        guest_keypair = BootstrapSelect('hardware__guest_access_key_pair')
+        hardware_monitoring = BootstrapSelect('hardware__monitoring')
+        boot_disk_size = BootstrapSelect('hardware__boot_disk_size')
+        # GCE
+        is_preemtible = VersionPick({
+            Version.lowest(): None,
```
Comment (mfalesni) [inline cfme/provisioning.py]: I am not sure how that is gonna be handled. It might need some more investigation :)

**Pull request merged**